src/iter.rs

```diff
@@ -48,3 +48,46 @@ impl PyIterProtocol for PyEmbeddingIterator {
         }
     }
 }
+
+#[pyclass(name=EmbeddingWithNormIterator)]
+pub struct PyEmbeddingWithNormIterator {
```

Comment:
I think this Iterator is not exposed anywhere in the Python package, do we want to expose this via an explicit `iter_with_norms()` method? Or should this iterator replace the `PyEmbeddingIterator` and be the default iterator over embeddings?

Reply:
`iter_with_norms()`, good catch!